rcl/test/CMakeLists.txt

```diff
@@ -1,117 +1,86 @@
 find_package(ament_cmake_gtest REQUIRED)
+find_package(std_msgs REQUIRED)
+
+include(rcl_add_custom_gtest.cmake)
 
 set(extra_test_libraries)
-set(extra_memory_tools_env PHONY=)  # Use a phony env var so there is always at least one.
-# rcl_lib_dir is passed as APPEND_LIBRARY_DIRS for each ament_add_gtest call so
-# the librcl that they link against is on the library path.
-# This is especially improtant on Windows.
-set(rcl_lib_dir $<TARGET_FILE_DIR:${PROJECT_NAME}>)
-ament_find_gtest()  # For GTEST_LIBRARIES
-if(APPLE)
-  add_library(${PROJECT_NAME}_memory_tools_interpose SHARED memory_tools_osx_interpose.cpp)
-  target_link_libraries(${PROJECT_NAME}_memory_tools_interpose ${GTEST_LIBRARIES})
-  set_target_properties(${PROJECT_NAME}_memory_tools_interpose
-    PROPERTIES COMPILE_FLAGS "-std=c++11")
-  list(APPEND extra_test_libraries ${PROJECT_NAME}_memory_tools_interpose)
-  list(APPEND extra_memory_tools_env
-    DYLD_INSERT_LIBRARIES=$<TARGET_FILE:${PROJECT_NAME}_memory_tools_interpose>)
-endif()
-add_library(${PROJECT_NAME}_memory_tools SHARED memory_tools.cpp)
-if(NOT WIN32)
-  set_target_properties(${PROJECT_NAME}_memory_tools PROPERTIES COMPILE_FLAGS "-std=c++11")
-endif()
-if(UNIX AND NOT APPLE)
-  list(APPEND extra_test_libraries dl)
-  list(APPEND extra_memory_tools_env DL_PRELOAD=$<TARGET_FILE:${PROJECT_NAME}_memory_tools>)
-endif()
-target_link_libraries(${PROJECT_NAME}_memory_tools ${extra_test_libraries})
-target_compile_definitions(${PROJECT_NAME}_memory_tools
-  PRIVATE "RCL_MEMORY_TOOLS_BUILDING_DLL")
-list(APPEND extra_test_libraries ${PROJECT_NAME}_memory_tools)
+set(extra_test_env PHONY=)  # Use a phony env var so there is always at least one.
+
+# This subdirectory extends both extra_test_libraries and extra_test_env.
+add_subdirectory(memory_tools)
 
-if(NOT WIN32)
-  ament_add_gtest(test_memory_tools test_memory_tools.cpp
-    ENV ${extra_memory_tools_env} APPEND_LIBRARY_DIRS ${rcl_lib_dir})
-  if(TARGET test_memory_tools)
-    target_include_directories(test_memory_tools PUBLIC
-      ${rcl_interfaces_INCLUDE_DIRS}
-      ${rmw_INCLUDE_DIRS}
-    )
-    if(NOT WIN32)
-      set_target_properties(test_memory_tools PROPERTIES COMPILE_FLAGS "-std=c++11")
-    endif()
-    target_link_libraries(test_memory_tools ${PROJECT_NAME} ${extra_test_libraries})
+macro(test_target)
+  # TODO(wjwwood): remove these when C type support is supported by typesupport_introspection.
+  if("${rmw_implementation} " STREQUAL "rmw_connext_dynamic_cpp ")
+    message(STATUS "Skipping tests for '${rmw_implementation}'")
+    return()
```

Comment:
That is a pretty bad side effect then. I would recommend change it to something more obvious.